### PR TITLE
lgc: set scalarizer min-bits to 32

### PR DESCRIPTION
**Pull request by nhaehnle**
Do not break up packed 16-bit math.

The test changes are admittedly a bit haphazard, but their point is primarily to focus the testing on how the frontend lowers SPIR-V to LGC, and to make the tests independent of future decisions regarding how and whether exactly we do scalarization.

This depends on https://reviews.llvm.org/D149842 (merged)

### CODE REVIEW

**Comment by nhaehnle** [comment]
Note: This will fail to build until the dependent LLVM change has propagate to our branch here.

**Comment by amdvlk-admin** [comment]
930f1ced1a533e39a532ad9c73ef1eb6b72212be Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/patch/Patch.cpp:383:21: error: ‘struct llvm::ScalarizerPassOptions’ has no member named ‘ScalarizeMinBits’; did you mean ‘ScalarizeLoadStore’?
   scalarizerOptions.ScalarizeMinBits = 32;
                     ^~~~~~~~~~~~~~~~
                     ScalarizeLoadStore
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/patch/Patch.cpp:384:47: error: no matching function for call to ‘llvm::ScalarizerPass::ScalarizerPass(llvm::ScalarizerPassOptions&)’
   fpm.addPass(ScalarizerPass(scalarizerOptions));
                                               ^
In file included from /jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/patch/Patch.cpp:96:0:
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llvm-project/llvm/include/llvm/Transforms/Scalar/Scalarizer.h:37:7: note: candidate: constexpr llvm::ScalarizerPass::ScalarizerPass()
 class ScalarizerPass : public PassInfoMixin<ScalarizerPass> {

**Comment by amdvlk-admin** [comment]
### Test summary for commit 27968e78a79b752b771c71b4e9aa092dec56b81f
<details><summary>CTS tests (Failed: 0/140881)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39058/70446 (55.4%)</li>
<li>Failed: 0/70446 (0.0%)</li>
<li>Not Supported: 31388/70446 (44.6%)</li><li>Warnings: 0/70446 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39090/70435 (55.5%)</li>
<li>Failed: 0/70435 (0.0%)</li>
<li>Not Supported: 31345/70435 (44.5%)</li><li>Warnings: 0/70435 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by nhaehnle** [comment]
ping

File: llpc/test/shaderdb/core/OpFOrdGreaterThanEqual_TestFloat_lit.frag [hunk 1]
```diff
@@ -20,10 +20,6 @@ void main()
 ; SHADERTEST: icmp uge <4 x i32>
 ; SHADERTEST: fcmp ult <4 x float>
 ; SHADERTEST: icmp sge <4 x i32>
-; SHADERTEST-LABEL: {{^// LLPC}}  pipeline patching results
-; SHADERTEST-COUNT-4: icmp uge
-; SHADERTEST-COUNT-4: fcmp ult
-; SHADERTEST-COUNT-4: icmp sge
```
[the remark this answers is not in the document]
Comment (nhaehnle) [reply llpc/test/shaderdb/core/OpFOrdGreaterThanEqual_TestFloat_lit.frag:26]: Yes, this was intentional. At least to me it seemed like checking for degree of scalarization is not something that ought to be done in these LLPC tests.

**Comment by nhaehnle** [comment]
cc @amdrexu

**Comment by piotrAMD** [comment]
I can see quite a lot of v_pk* instructions generated as a result of this patch.

**Review by piotrAMD — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Pull request merged**